app/controllers/application_controller.rb

```diff
@@ -1,4 +1,31 @@
 class ApplicationController < ActionController::Base
   include AllowBrowser, Authentication, Authorization, SetCurrentRequest, SetPlatform, TrackedRoomVisit, VersionHeaders
   include Turbo::Streams::Broadcasts, Turbo::Streams::StreamName
+
+  before_action :forbid_banned_user
+
+  private
+    def forbid_banned_user
+      return unless Current.user&.banned?
+      return if banned_user_allowed_action?
+
+      # GG: flash seems is unable to be shown, so we do not put it
+      # flash.now[:alert] = "Your account has been banned."
+
+      if turbo_request?
+        render turbo_stream: turbo_stream.update("main-content", partial: "application/banned_notice"), status: :forbidden
+      elsif request.format.html?
+        render "application/banned", status: :forbidden
+      else
+        head :forbidden
+      end
```

Comment:
Did you consider revoking the user's session when they get banned?
That way they'd be forced to re-login and you wouldn't have to handle all these cases.

Reply:
Hum my (humble) idea was to try to show something to the user and let him log in.
Also if the user is unbanned, (s)he get access back immediatly.

Reply:
From my perspective, you are banning someone because you don't want them on your instance. So you want to completely forbid them from using any resources on your instance. And session do use up some resources occasionally. You can still let the user know that they've been banned with a flash message in the Sessions controller.